### PR TITLE
Show alert if map cannot load

### LINKED ISSUE #8230: Inform users they need to unblock 3rd party browser cookies in Firefox to make /map work
## Description
Currently if user has disabled 3rd party cookies, for example in Firefox:
image

When the user goes to /map the map wont show correctly
image

And bugsnagJS errors will be created like:
https://app.bugsnag.com/yaycode/open-food-network-js-katuma/errors/5e8c6364df4e9c0017b280c6?filters[event.since][0]=30d&filters[error.status][0]=open
image

But the user is not informed about this error in any way. So for the user, the OFN map is broken.

We could make a very simple change in OFN to detect this problem and show an error message to the user indicating they should enable third-party cookies if they want to see the OFN map.

All occurrences of this error in bugsnagJS (121 in France) are on Firefox, so this is a Firefox specific issue. The map works correctly in chrome even if 3rd party cookies are disabled.

## Acceptance Criteria & Tests
Users who have disabled 3rd party cookies are warned when they visit /map that they can enable 3rd party cookies to see the map.

### PR DESCRIPTION
**Pull request by murjax**
#### What? Why?

- Closes #8230

Some browsers may fail to load Google Maps if 3rd party cookies are disabled. This causes downstream errors where the search bar attempts to render. More importantly, the user doesn't know why the map might have failed to load. This solution shows an alert instructing the user to check their network and 3rd party cookie settings.

#### What should we test?
Visit `/map`. Ensure the map still loads correctly.

The 3rd party cookie issue cannot be directly reproduced. Instead, the page has to be loaded while offline. This can only be done in local development where the page is locally hosted and Google Maps is loaded from the network.

#### Release notes

Changelog Category (reviewers may add a label for the release notes):

- [x] User facing changes
- [ ] API changes (V0, V1, DFC or Webhook)
- [ ] Technical changes only
- [ ] Feature toggled

The title of the pull request will be included in the release notes.

### CODE REVIEW

**Review by chahmedejaz — CHANGES_REQUESTED** [review]
Great work @murjax. We are almost there, we just need to address a couple of robocop lint warnings. Thanks. 😄 

Lint Warnings:
https://github.com/openfoodfoundation/openfoodnetwork/actions/runs/11632701992/job/32396353444?pr=12963#step:5:74

**Comment by murjax** [comment]
@chahmedejaz Thanks! Lint warnings are fixed.

**Review by chahmedejaz — APPROVED** [review]
LGTM. Thanks 👍

File: config/locales/en.yml [hunk 1]
```diff
@@ -2784,6 +2784,7 @@ See the %{link} to find out more about %{sitename}'s features and to start using
   confirm_hub_change: "Are you sure? This will change your selected hub and remove any items in your shopping cart."
   confirm_oc_change: "Are you sure? This will change your selected order cycle and remove any items in your shopping cart."
   location_placeholder: "Type in a location..."
+  gmap_load_failure: "Unable to load map. Please check your network and allow 3rd party cookies."
```
Comment (rioug) [inline config/locales/en.yml:2787]: I think the error message is a bit confusing, maybe something like this would be better :
"Unable to load map. Please check your browser settings and allow 3rd party cookies for this website"

Comment (murjax) [reply config/locales/en.yml]: Works for me. I made the change.

**Review by chahmedejaz — APPROVED** [review]

File: spec/system/consumer/map_spec.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+      end
+
+      page.driver.browser.url_whitelist = url_whitelist
+    end
+  end
+
+  context 'map cannot load' do
+    it 'shows alert' do
+      message = accept_alert { visit '/map' }
+      expect(message).to eq(I18n.t('gmap_load_failure'))
```
Comment (rioug) [inline spec/system/consumer/map_spec.rb:22]: As a general rule we prefer to check with the actual message, so we can check the translation is working ,ie: 
```suggestion
      expect(message).to eq("Unable to load map. Please check your browser settings and allow 3rd party cookies for this website.")
```

**Review by rioug — CHANGES_REQUESTED** [review]

File: spec/system/consumer/map_spec.rb (hunk 2 above)
Comment (murjax) [reply spec/system/consumer/map_spec.rb]: The string needs to be broken across multiple lines in this case to keep rubocop happy. I pushed a change that uses `String#squish`, but let me know if you have a preferred method for this.

**Review by rioug — APPROVED** [review]
Thanks @murjax Looks good now!
For future reference, once a PR has been reviewed it's better to add new commit instead updating existing commit. Otherwise we loose the review history and it can get confusing for other reviewer.

File: spec/system/consumer/map_spec.rb (hunk 2 above)
Comment (rioug) [reply spec/system/consumer/map_spec.rb]: I personally prefer to split the string with a `\` but `squish` work as well. ie:
```ruby
 expect(message).to eq(
  "Unable to load map. Please check your browser" \
  " settings and allow 3rd party cookies for this website."
)

Comment (murjax) [reply spec/system/consumer/map_spec.rb]: I like this better. I committed this changed.

**Review by rioug — APPROVED** [review]

**Comment by filipefurtad0** [comment]
Hey @murjax ,

Apologies, I'm not very sure how to test this PR. Indeed, I could not reproduce issue https://github.com/openfoodfoundation/openfoodnetwork/issues/8230, I've disabled third-party cookies in Firefox/Chrome but the map still loaded.

Also, locally, the map was loading even though third party cookies were disabled.

I think I might be missing something obvious; what would be the best way to test your code changes?

Thanks for your help.

**Comment by murjax** [comment]
@filipefurtad0 
The third-party cookie path is not reproducible.

However, the TypeError error mentioned in the description of #8230 was reproducible by going offline and reloading the map page locally. The assumption here is that third-party cookies being disabled prevents the map resources from loading on some browsers.

With this fix, reloading the map page while offline should present the alert indicating the map was unable to load and prompting the user to check their settings.

To clarify, this offline scenario won't occur in a staging or production environment because the page can't load to begin with while offline. It's possible in local development though because the app is hosted locally but pulls the map resources from the network.

**Comment by filipefurtad0** [comment]
Right, thanks for your help @murjax , I was able to spot this now on my local dev environment, when offline:

Master branch - no warning is visible, just the map not showing up
image

Your branch - warning is visible
image

Since we don't have system tests on the maps feature, I've staged the PR and did some smoke tests to check basic functionality, for instance, that open shops are displayed on the map according to the given address.

This looks good, merging.
Thanks again! :muscle:

**Comment by mkllnk** [comment]
One of the specs is failing since being merged to master:
```
Failures:

  1) Map map can load does not show alert
     Failure/Error:
       assert_raises(Capybara::ModalNotFound) do
         accept_alert { visit '/map' }
       end
     
     Minitest::Assertion:
       Capybara::ModalNotFound expected but nothing was raised.
     
     [Screenshot Image]: /home/runner/work/openfoodnetwork/openfoodnetwork/tmp/capybara/screenshots/failures_r_spec_example_groups_map_map_can_load_does_not_show_alert_64.png

     
     # ./spec/system/consumer/map_spec.rb:11:in `block (3 levels) in <top (required)>'
     # ./spec/system/support/cuprite_setup.rb:39:in `block (2 levels) in <top (required)>'
     # ./spec/base_spec_helper.rb:153:in `block (3 levels) in <main>'
     # ./spec/base_spec_helper.rb:153:in `block (2 levels) in <main>'
```
